scripts/context_retrieval/build_database.py

```diff
@@ -0,0 +1,652 @@
+"""
+Downloads and persists SQuAD training data. Builds and persists LlamaIndex database. Computes OpenAI
+embeddings for query and database data.
+"""
+
+import json
+import logging
+import os
+import sys
+import urllib
+from typing import Any, List, Tuple, TypedDict
+
+import numpy as np
+import openai  # type: ignore
```

Comment:
can't you add this as a dev deps to satisfy the type hints or do they just not exist?

Reply:
I can add these as dev deps.